src/codegen/primitive_value_proxy.cpp

```diff
@@ -0,0 +1,325 @@
+
+const std::string &PrimitiveValueProxy::_GetInteger::GetFunctionName() {
+  static const std::string kGetIntegerFnName =
+      "_ZN7peloton7codegen19PrimitiveValueProxy10GetIntegerEPPcj";
+  return kGetIntegerFnName;
+}
+
+llvm::Function *PrimitiveValueProxy::_GetInteger::GetFunction(
+    CodeGen &codegen) {
+  const std::string &fn_name = GetFunctionName();
+
+  // Has the function already been registered?
+  llvm::Function *llvm_fn = codegen.LookupFunction(fn_name);
+  if (llvm_fn != nullptr) {
+    return llvm_fn;
+  }
+
+  auto *fn_type = llvm::FunctionType::get(
+      codegen.Int32Type(),
+      {codegen.CharPtrType()->getPointerTo(), codegen.Int64Type()},
+      false);
+  return codegen.RegisterFunction(fn_name, fn_type);
+}
+
+//===----------------------------------------------------------------------===//
+// GET BIGINT
+//===----------------------------------------------------------------------===//
+
+const std::string &PrimitiveValueProxy::_GetBigInt::GetFunctionName() {
+  static const std::string kGetBigIntFnName =
+      "_ZN7peloton7codegen19PrimitiveValueProxy9GetBigIntEPPcj";
+  return kGetBigIntFnName;
+}
+
+llvm::Function *PrimitiveValueProxy::_GetBigInt::GetFunction(
+    CodeGen &codegen) {
+  const std::string &fn_name = GetFunctionName();
+
+  // Has the function already been registered?
+  llvm::Function *llvm_fn = codegen.LookupFunction(fn_name);
+  if (llvm_fn != nullptr) {
+    return llvm_fn;
+  }
+
+  auto *fn_type = llvm::FunctionType::get(
+      codegen.Int64Type(),
+      {codegen.CharPtrType()->getPointerTo(), codegen.Int64Type()},
+      false);
+  return codegen.RegisterFunction(fn_name, fn_type);
+}
+
+//===----------------------------------------------------------------------===//
+// GET DOUBLE
+//===----------------------------------------------------------------------===//
+
+const std::string &PrimitiveValueProxy::_GetDouble::GetFunctionName() {
+  static const std::string kGetDoubleFnName =
+      "_ZN7peloton7codegen19PrimitiveValueProxy9GetDoubleEPPcj";
+  return kGetDoubleFnName;
+}
+
+llvm::Function *PrimitiveValueProxy::_GetDouble::GetFunction(
+    CodeGen &codegen) {
+  const std::string &fn_name = GetFunctionName();
+
+  // Has the function already been registered?
+  llvm::Function *llvm_fn = codegen.LookupFunction(fn_name);
+  if (llvm_fn != nullptr) {
+    return llvm_fn;
+  }
+
+  auto *fn_type = llvm::FunctionType::get(
+      codegen.DoubleType(),
+      {codegen.CharPtrType()->getPointerTo(), codegen.Int64Type()},
+      false);
+  return codegen.RegisterFunction(fn_name, fn_type);
+}
+
+//===----------------------------------------------------------------------===//
+// PEEK TIMESTAMP
+//===----------------------------------------------------------------------===//
+
+const std::string &PrimitiveValueProxy::_GetTimestamp::GetFunctionName() {
+  static const std::string kGetTimestampFnName =
+#ifdef __APPLE__
+        "_ZN7peloton7codegen13ValuesRuntime13outputVarcharEPcjS2_j";
```

Comment:
I understand your argument, and appreciate for bringing it up.  I think it would be good to think that we write code with supporting OSX in mind. Although it is broken now, we supported OSX last year, and we are still expecting to fix it in very near future. 

The best solution to this is to apply the correct name mangling for OSX, but for the time being we choose to apply the same mangled name as a NON-APPLE one, throughout the codegen code. I think it would be a good practice to keep this strategy. Just be kind to the engineers who might be working on it in near future.  You could choose the best solution, though.

The more important problem here is the inconsistencies in handling these name mangling's throughout this PR: some have no APPLE macro, some others have APPLE macro with the same name as non-apple's, and the others have APPLE macro with some other names.  I think this is not what you want to do as an engineer in this kind of big project.